### ISSUE TITLE
Redis Health Check

### ISSUE THREAD

**Issue by RehanSaeed** [Xabaril/AspNetCore.Diagnostics.HealthChecks#750]
**What would you like to be added**:

I have already registered a singleton `IConnectionMultiplexer` to my IoC container and I would like my health check to use that instead.

```
// Func<IServiceProvider, IConnectionMultiplexer>
.AddRedis(sp => sp.GetRequiredService<IConnectionMultiplexer>())
```

**Why is this needed**:

To use the same connection as the application

**Comment by runes83** [comment]
I would also need this because in GCP you have to add custom CA certificate, this cannot be done in the connectionstring have to be setupin the ConnectionMultiplexer

**Comment by CleytonGoncalves** [comment]
Yeah, +1. Several cloud providers might require custom CA certificate.

**Comment by sungam3r** [comment]
PR is welcome.
